### ISSUE TITLE
Algolia search is broken

### ISSUE THREAD

**Issue by baumannzone** [vuejs/vitepress#1970]
### Describe the bug

I am experiencing an issue with the integrated search component (Algolia).
The search function is not working properly, as it returns **zero results** when I use the web interface in my recently created project with Vitepress. 

However, if I perform a search using the API or on Algolia's website, the search function works correctly and returns results.

It's possible that there may be an issue with the search component in Vitepress that's causing this problem. 

I wanted to bring this to your attention in the hopes that it can be addressed and fixed as soon as possible. 

1. Please let me know if there is any further information or details I can provide to assist with troubleshooting this issue."

### Reproduction

- Create an Algolia account
- Configure the algolia section with index, API key and app id.
  - ```
      appId: '5D6FRGxxxx',  
      apiKey: 'e04451b589c484090xxxxxxxxxxxxxxx',  
      indexName: 'csspills'
      ```
- Create the index (both manually / via API)
  - ```
    [
      {
        title: 'Pill 1 CLI',
        description: 'Create a new project with the CLI',
        tags: [ 'demo1', 'cli' ]
      },
      {
        title: 'Pill 2 AWS',
        description: 'Create a new AWS project from the command line',
        tags: [ 'demo2', 'AWS' ]
      }
    ]
    ```
- Start the local server
- Search 

![image](https://user-images.githubusercontent.com/5422102/220378526-6c84ae52-ccab-4270-b25f-73f8135e60b5.png)

### Expected behavior

Show results after searching

### System Info

```shell
System:
    OS: macOS 13.2.1
    CPU: (8) x64 Apple M1
    Memory: 7.73 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Browsers:
    Brave Browser: 107.1.45.133
    Chrome: 109.0.5414.119
    Chrome Canary: 112.0.5609.0
    Firefox: 95.0.2
    Firefox Developer Edition: 108.0
    Safari: 16.3
```

### Additional context

Request URL: 
```
https://5d6frgc6t9-dsn.algolia.net/1/indexes/*/queries?x-algolia-agent=Algolia%20for%20JavaScript%20(4.8.5)%3B%20Browser%20(lite)%3B%20docsearch%20(3.3.3)%3B%20docsearch-react%20(3.3.3)%3B%20docsearch.js%20(3.3.3)&x-algolia-api-key=e04451b589c484090xxxxxxxxxxxxxxx&x-algolia-application-id=5D6FRGxxxx
```
Request Method: `POST`
Status Code: `200 OK`

![image](https://user-images.githubusercontent.com/5422102/220379976-4dda3eff-b8ab-4b63-98b0-6282d58bcef5.png)

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.vuejs.org).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by baumannzone** [comment]
Please let me know if there is any further information or details I can provide to assist with troubleshooting this issue."

**Comment by brc-dd** [comment]
your index needs the `lang` information too as we are filtering based on that

**Comment by baumannzone** [comment]
Hi, @brc-dd! Thanks for your reply.
Where should I put that `lang` key and what values does it accept, any link to the doc?

My question is, why does it work through the API?

**Comment by brc-dd** [comment]
I'm not sure how you're creating that index, but if you'd run algolia's crawler it would've indexed the html lang attribute too. 

It's working through the API because you haven't specified `facetFilters: ['lang:en-US']` in your request.

**Comment by baumannzone** [comment]
Im using the free quote, I don't have access to the crawler. I'm completely new to Algolia, I may be missing something (?)

**Comment by brc-dd** [comment]
Just a clarification on our side: we only support https://docsearch.algolia.com/ . If that's the case for you, then you can use https://crawler.algolia.com/admin/ to access crawler configuration.

**Comment by baumannzone** [comment]
Thanks for your clarification. It's a public repo and it's open source. 
Here is [the website](https://onebeyond.github.io/css-pills/) and this is [the repo](https://github.com/onebeyond/css-pills).

**Comment by brc-dd** [comment]
Yeah, you should have, can you try logging in that admin dashboard?

**Comment by baumannzone** [comment]
Thank you. Actually, I dont have access. But it looks like an auth error. I will try later.

**Comment by baumannzone** [comment]
I just requested access to the crawler. It seems to be that was the problem.
If that doesn't solve my problem, I'll be creating a custom search component. Let's keep this issue open until I figure out what's going on.

As feedback, I think Vitepress docs should provide more context about how Algolia should work.

**Comment by MarkusKeck** [comment]
If you really need a fast solution and you need the newest vitepress version you can try `vitepress-plugin-search`

```ts
//vite.config.ts
import { SearchPlugin } from "vitepress-plugin-search";
import { defineConfig } from "vite";

//default options
var options = {
  ...flexSearchIndexOptions,
  previewLength: 62,
  buttonLabel: "Search",
  placeholder: "Search docs",
};

export default defineConfig({
  plugins: [SearchPlugin(options)],
});
```

**Comment by brc-dd** [comment]
> As feedback, I think Vitepress docs should provide more context about how Algolia should work.

Yeah, the docs are missing many things. I agree that it's sort of ironic considering we are building a docs-centered framework 😅. But yeah we will update that.

After you get access, try adding attributesForFaceting (ref. https://github.com/vuejs/vitepress/issues/1935#issuecomment-1433765099)

**Comment by baumannzone** [comment]
Thank you very mucho, amigo. I'll let you know how it goes. 

@MarkusKeck I want to try using the default search component, I'll try your solution if I can't get it to work.

**Comment by baumannzone** [comment]
I just got a message, it says my site won't be accepted bcs is under construction and has tons of `lorem ipsum`s text. But I think when I finish creating the website, I will be given access to the tool.

**Comment by brc-dd** [comment]
Let's move this to discussion. If there is any issue you need some help you can directly post in our discord too.
